core/src/main/java/org/apache/iceberg/SortOrderParser.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Iterator;
+import java.util.Locale;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.JsonUtil;
+
+public class SortOrderParser {
+  private static final String ORDER_ID = "order-id";
+  private static final String FIELDS = "fields";
+  private static final String DIRECTION = "direction";
+  private static final String NULL_ORDERING = "null-order";
+  private static final String TRANSFORM = "transform";
+  private static final String SOURCE_ID = "source-id";
+
+  private SortOrderParser() {
+  }
+
+  public static void toJson(SortOrder sortOrder, JsonGenerator generator) throws IOException {
+    generator.writeStartObject();
+    generator.writeNumberField(ORDER_ID, sortOrder.orderId());
+    generator.writeFieldName(FIELDS);
+    toJsonFields(sortOrder, generator);
+    generator.writeEndObject();
+  }
+
+  private static void toJsonFields(SortOrder sortOrder, JsonGenerator generator) throws IOException {
+    generator.writeStartArray();
+    for (SortField field : sortOrder.fields()) {
+      generator.writeStartObject();
+      generator.writeStringField(TRANSFORM, field.transform().toString());
```

Comment:
Do we need to write a transform if it is identity? It would be nice to omit this when it is identity.

Reply:
I am ok omitting but it will add extra logic and it will behave differently compared to `PartitionSpec`.